### PR TITLE
Bug 1102675 - prevent app from opening camera hw while a sharing activity is active

### PR DESCRIPTION
**Pull request by mikeaich**
...ve

### CODE REVIEW

**Comment by try-server-hook** [comment]
[mikeaich Mike Habicher (mikeaich)](https://github.com/mikeaich) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=88e29f7fd271)

**Comment by try-server-hook** [comment]
[mikeaich Mike Habicher (mikeaich)](https://github.com/mikeaich) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8d30c6c2245d)

File: apps/camera/js/app.js [hunk 1]
```diff
@@ -499,4 +500,28 @@ App.prototype.listenForStopRecordingEvent = function() {
   addEventListener('stoprecording', this.firer('stoprecording'));
 };
 
+App.prototype.isSharingActive = function() {
+  debug('is sharing active?');
+  return this._sharingIsActive;
+};
+
+App.prototype.setSharingState = function(state) {
+  var isActive = state === 'sharing' ? true : false;
```
Comment (justindarc) [inline apps/camera/js/app.js]: Can simply be:

```
var isActive = state === 'sharing';
```

File: apps/camera/js/app.js [hunk 2]
```diff
@@ -499,4 +500,28 @@ App.prototype.listenForStopRecordingEvent = function() {
...
+  if (isActive == this._sharingIsActive) {
```
Comment (justindarc) [inline apps/camera/js/app.js]: Should be `===`

**Comment by try-server-hook** [comment]
[mikeaich Mike Habicher (mikeaich)](https://github.com/mikeaich) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=387cceddcf3b)

**Pull request merged**